README.md

```diff
@@ -679,6 +679,61 @@ It also means that these rules will match against following two events :
 
 This behaviour may change in future version (to avoid any confusions) and should not be relied upon.
 
+#### Caveat: JSON parsing and duplicate keys
+
+Events or Rules containing duplicate JSON keys are invalid. When duplicate keys are passed Ruler 
+will only consider the final value in the document order. 
+
+This means following two rules will match the same events.
+
+```javascript
+## has duplicate keys
+{
+  "source": ["aws.s3"],
+  "source": ["aws.sns"],
+  "detail-type": ["AWS API Call via CloudTrail"],
+  "detail":  {
+      "eventSource": ["s3.amazonaws.com"],
+      "eventSource": ["sns.amazonaws.com"]
+  }
+}
+
+## has unique keys
+{
+  "source": ["aws.sns"],
+  "detail-type": ["AWS API Call via CloudTrail"],
+  "detail": { "eventSource": ["sns.amazonaws.com"] }
+}
+```
+
+Similarly, following two *events* will be treated the same
+
+```javascript
+## has duplicate keys
+{
+  "source": ["aws.s3"],
+  "source": ["aws.sns"],
+  "detail-type": ["AWS API Call via CloudTrail"],
+  "detail":  [
+    {
+      "eventSource": ["s3.amazonaws.com"],
+      "eventSource": ["sns.amazonaws.com"]
+    }
+  ]
+}
+
+## has unique keys
+{
+  "source": ["aws.sns"],
+  "detail-type": ["AWS API Call via CloudTrail"],
+  "detail": [
+    { "eventSource": ["sns.amazonaws.com"] }
+  ]
+}
+```
+
+This behaviour will change in future version (to avoid any confusions) and should not be relied upon.
```

Comment:
"Will"? As in we will definitely change the behavior? For backwards compatibility, I'm pretty sure we will keep most of this behavior the same, with the possible exception or rejecting rules (but not events) with duplicate keys. The word "may" is probably more appropriate than "will".